### PR TITLE
Add support for elogind

### PR DESCRIPTION
**Pull request by shibotto**
Support for elogind, as already reported here: https://github.com/cmus/cmus/issues/844

### CODE REVIEW

**Comment by flyingmutant** [comment]
Sorry for such a delayed reply. Aren't elogind headers in `<elogind/...>` instead of `<systemd/...>`?

**Comment by shibotto** [comment]
It's OK :D
Actually elogind is meant to be a drop-in replacement of systemd's logind. For that purpose all the headers are stored in `/usr/include/elogind/systemd/*` and [pkg-config](https://github.com/elogind/elogind/blob/ec32681ee46e16f1dad6848f5a83583bc550805d/src/libelogind/libelogind.pc.in) takes care of importing the right directory, so the source code doesn't need to be altered.

**Comment by flyingmutant** [comment]
I have a feeling that it is better to limit this patch to just this line: https://github.com/cmus/cmus/pull/846/files#diff-e2d5a00791bce9a01f99bc6fd613a39dR292, because AFAIK there is nothing really logind-specific in the APIs cmus uses, they are just a part of systemd's sd-bus. @mahkoh any thoughts?

**Comment by nefthy** [comment]
Patch LGTM

**Comment by geek1011** [comment]
@flyingmutant Can this be merged either way?

File: configure [hunk 1]
```diff
@@ -287,9 +287,9 @@ check_vorbis()
 	fi
 }
 
-check_libsystemd()
+check_logind()
 {
-	pkg_config LIBSYSTEMD "libsystemd"
+	pkg_config LOGIND "libsystemd" || pkg_config LOGIND "libelogind >= 239.3"
```
Comment (flyingmutant) [inline configure]: Can we only leave this bit in the patch? So that `LIBSYSTEMD_` stays etc.? I don't like having `elogind` names around.

Comment (geek1011) [reply configure]: Yes, that should work.

Comment (shibotto) [reply configure]: I just wanted to let you discuss this "internally" first, but I actually don't mind 😀

I'm not quite fond myself of the `LOGIND` change, but `LIBSYSTEMD` seemed even more generic, as only sd-bus is required. Anyway, pushing this!

Comment (nefthy) [reply configure]: Why libelogind >= 239.3? Are earlier versions not compatible?

Comment (shibotto) [reply configure]: @nefthy if I remember correctly version previous to 239.3 and 238.3 didn't have complete sd-bus support. I'm not sure how to correctly address this scenario.

Comment (geek1011) [reply configure]: Yes, that's correct, but AFAIK, it should work fine quite a few versions back too.

Comment (nefthy) [reply configure]: I'm OK with the check. Just wanted to make sure it is not arbitrary.

**Comment by nefthy** [comment]
There is something wrong here. There are changes in the diff, that stem from other PRs. It seems that upstream was merged into this branch instead of rebasing the branch on top of master.

please rebase.

**Comment by shibotto** [comment]
Sorry, I wanted to finish this before going out this morning, but doing things in a hurry is never a good a idea :/

**Comment by nefthy** [comment]
Looks good. Thanks for your work.

**Pull request merged**